src/ufo.rs

```diff
@@ -69,28 +72,352 @@ impl Default for MetaInfo {
+    pub open_type_vhea_vert_typo_line_gap: Option<i32>,
+    pub postscript_blue_fuzz: Option<f64>,
+    pub postscript_blue_scale: Option<f64>,
+    pub postscript_blue_shift: Option<f64>,
+    pub postscript_blue_values: Option<Vec<f64>>,
+    pub postscript_default_character: Option<String>,
+    pub postscript_default_width_x: Option<f64>,
+    pub postscript_family_blues: Option<Vec<f64>>,
+    pub postscript_family_other_blues: Option<Vec<f64>>,
+    pub postscript_font_name: Option<String>,
+    pub postscript_force_bold: Option<bool>,
+    pub postscript_full_name: Option<String>,
+    pub postscript_is_fixed_pitch: Option<bool>,
+    pub postscript_nominal_width_x: Option<f64>,
+    pub postscript_other_blues: Option<Vec<f64>>,
+    pub postscript_slant_angle: Option<f64>,
+    pub postscript_stem_snap_h: Option<Vec<f64>>, // TODO: validate. (i32|f64)?
+    pub postscript_stem_snap_v: Option<Vec<f64>>, // TODO: validate. (i32|f64)?
+    pub postscript_underline_position: Option<f64>,
+    pub postscript_underline_thickness: Option<f64>,
+    #[serde(rename = "postscriptUniqueID")]
+    pub postscript_unique_id: Option<i32>,
+    pub postscript_weight_name: Option<String>,
+    pub postscript_windows_character_set: Option<PostscriptWindowsCharacterSet>,
     pub style_map_family_name: Option<String>,
-    pub style_map_style_name: Option<String>,
-    pub version_major: Option<u32>,
-    pub version_minor: Option<u32>,
-    pub year: Option<u32>,
-    pub copyright: Option<String>,
+    #[serde(deserialize_with = "deserialize_style_map_style_name")]
+    pub style_map_style_name: Option<StyleMapStyle>,
+    pub style_name: Option<String>,
     pub trademark: Option<String>,
     pub units_per_em: Option<f64>,
-    pub descender: Option<f64>,
+    pub version_major: Option<u32>,
+    pub version_minor: Option<u32>,
+    pub woff_major_version: Option<i32>,
+    pub woff_metadata_copyright: Option<WoffMetadataCopyright>,
+    pub woff_metadata_credits: Option<WoffMetadataCredits>,
+    pub woff_metadata_description: Option<WoffMetadataDescription>,
+    pub woff_metadata_extensions: Option<Vec<WoffMetadataExtensionRecord>>,
+    pub woff_metadata_license: Option<WoffMetadataLicense>,
+    pub woff_metadata_licensee: Option<WoffMetadataLicensee>,
+    pub woff_metadata_trademark: Option<WoffMetadataTrademark>,
+    #[serde(rename = "woffMetadataUniqueID")]
+    pub woff_metadata_unique_id: Option<WoffMetadataUniqueID>,
+    pub woff_metadata_vendor: Option<WoffMetadataVendor>,
+    pub woff_minor_version: Option<i32>,
     pub x_height: Option<f64>,
-    pub cap_height: Option<f64>,
-    pub ascender: Option<f64>,
-    pub italic_angle: Option<f64>,
-    pub note: Option<String>,
+    pub year: Option<u32>,
+}
+
+// TODO: validate!
+// http://unifiedfontobject.org/versions/ufo3/fontinfo.plist/#guidelines
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct Guideline {
```

Comment:
this type already exists in `src/glyph/mod.rs`. Can we reuse it? We might want to move it to a different file as well, but for now I think it's fine to just leave it there.

Reply:
Oh, hm, maybe, if validation and de/serialization can be shared.